### PR TITLE
SCMOD-12730: Add index + drop context column

### PR DESCRIPTION
**Pull request by rorytorneymf**
https://portal.digitalsafe.net/browse/SCMOD-12730

### CODE REVIEW

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
I don't see the need to retain the `idx_partition_id_and_job_id` index given this one being added.

**Comment by rorytorneymf** [comment]
> I don't see the need to retain the `idx_partition_id_and_job_id` index given this one being added.

I think we still need this existing index to fetch **all** the stowed tasks for a job:

https://github.com/JobService/worker-taskunstowing/blob/SCMOD-12730/worker-taskunstowing/src/main/java/com/github/jobservice/workers/taskunstowing/TaskUnstowingWorker.java#L98

```
stowedTaskRows = databaseClient.getStowedTasks(partitionId, jobId);
```

This additional index would then be used to delete a specific stowed task, after it had been unstowed:

https://github.com/JobService/worker-taskunstowing/blob/SCMOD-12730/worker-taskunstowing/src/main/java/com/github/jobservice/workers/taskunstowing/TaskUnstowingWorker.java#L129

```
databaseClient.deleteStowedTask(partitionId, jobId, taskMessage.getTracking().getJobTaskId());
```

**Comment by dermot-hardy** [comment]
If you read up on the way PostgreSQL B-tree indexes work you should see why this change makes the `idx_partition_id_and_job_id` essentially redundant.

**Comment by rorytorneymf** [comment]
> If you read up on the way PostgreSQL B-tree indexes work you should see why this change makes the `idx_partition_id_and_job_id` essentially redundant.

Thanks, I've dropped the old 2-column index now, and also dropped the context column as well as discussed.

File: job-service-db/src/main/resources/v4.0.0/changelog.xml [hunk 1]
```diff
@@ -101,4 +101,38 @@
...
+    <changeSet id="drop_task_stowing_idx_partition_id_and_job_id" author="Gamma Team">
+        <preConditions onFail="CONTINUE">
+            <indexExists schemaName="public"
+                         tableName="stowed_task"
+                         indexName="idx_partition_id_and_job_id"
+                         columnNames="partition_id, job_id"/>
+        </preConditions>
+        <dropIndex indexName="idx_partition_id_and_job_id" tableName="stowed_task"/>
+    </changeSet>
+    <changeSet id="create_task_stowing_idx_partition_id_and_job_id_and_tracking_info_job_task_id" author="Gamma Team">
```
Comment (rorytorneymf) [inline job-service-db/src/main/resources/v4.0.0/changelog.xml:113]: Should I also add another changeSet to make this the primary key?

```
<changeSet id="add_task_stowing_primary_key" author="Gamma Team">
    <addPrimaryKey schemaName="public" tableName="stowedTask" columnNames="partition_id, job_id, tracking_info_job_task_id" />
</changeSet>
```

Comment (dermot-hardy) [reply job-service-db/src/main/resources/v4.0.0/changelog.xml:113]: That would have further implications for the calling code so you'd be better leaving that for another Jira.

**Pull request closed without merging**